federation-integration-testsuite-js/src/fixtures/accounts.ts

```diff
@@ -78,6 +78,7 @@ export const typeDefs = gql`
     id: ID!
     name: String @external
     userAccount(id: ID! = "1"): User @requires(fields: "name")
+    description: String
```

Comment:
Long story short, we implement Federation v1 internally and are planning to move to v2 and I'm facing the following issue; if an overridden field is on a type that implements an interface, composition will fail since removing `@join__field` from the supergraph sdl yields an invalid subgraph when extracting subgraphs from the  supergraph. 

To reproduce, we can add the following:

```
  type Library implements Describable @key(fields: "id") {
    ...
   }
  interface Describable {
    description: String
  }
```

and any tests in `buildQueryPlan.test.ts` will now fail with the following error:

```
Unexpected error extracting subgraph accounts from the supergraph: this is either a bug, or the supergraph has been corrupted..

    Details:
    [accounts] Interface field Describable.description expected but Library does not provide it.
    
    at extractSubgraphsFromSupergraph (../internals-js/src/extractSubgraphsFromSupergraph.ts:300:15)
      at buildFederatedQueryGraph (../query-graphs-js/src/querygraph.ts:523:51)
      at compose (../composition-js/src/compose.ts:53:55)
      at composeServices (../composition-js/src/compose.ts:85:10)
      at Object.<anonymous> (../federation-integration-testsuite-js/src/fixtures/special-cases/supergraphWithInaccessible.ts:13:42)
      at Object.<anonymous> (../federation-integration-testsuite-js/src/fixtures/index.ts:56:1)
```

I stumbled upon this PR and I'm really looking forward to the override functionality! Thank you for putting this one together 🙏

Reply:
Thanks, and you're right, we also need to handle interfaces here.

@clenfest: that's a case we forgot. One other reason why an overridden field may need to be kept external rather than removed is if that field exists to fulfil an interface. Like for key/provides/requires, we should detect it and mark the field external in the supergraph in that case.